raft/tracker/progress.go

```diff
@@ -55,9 +55,13 @@ type Progress struct {
 	// This is always true on the leader.
 	RecentActive bool
 
-	// ProbeSent is used while this follower is in StateProbe. When ProbeSent is
-	// true, raft should pause sending replication message to this peer until
-	// ProbeSent is reset. See ProbeAcked() and IsPaused().
+	// ProbeSent is true when a "probe" MsgApp was sent to this follower recently,
+	// and we haven't heard from it back yet. Used when the MsgApp flow is
+	// throttled, i.e. when State is StateProbe, or StateReplicate with saturated
+	// Inflights. In both cases, we need to continue sending MsgApp once in a
+	// while to guarantee progress, but we only do so when ProbeSent is false (it
+	// is reset on receiving a heartbeat response), to not overflow the receiver.
+	// See IsPaused().
 	ProbeSent bool
```

Comment:
`ProbeSent` is supposed to be used only for the `StateProbe`.

Based on the comment above, the `ProbeSent` doesn't mean exactly what its name indicates; it's also used for the `tracker.StateReplicate` (set to `true` when the Inflight is full). It's an anti-pattern to me. I would suggest to dedicate `ProbeSent` to what's its name indicates (preferred), otherwise we need to change the name (not preferred).

Reply:
Please always keep readability in mind, otherwise any new contributors might get confused. 

Good to see the new name, which is clearer.